src/EditorFeatures/Test2/GoToImplementation/GoToImplementationTests.vb

```diff
@@ -0,0 +1,150 @@
+        Public Sub TestInvalidInvocationSymbol()
+            Dim workspace =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document>
+class $$C { }
+        </Document>
+    </Project>
+</Workspace>
+
+            Test(workspace, shouldSucceed:=False)
+        End Sub
+
+        <Fact, Trait(Traits.Feature, Traits.Features.GoToImplementation)>
+        Public Sub TestWithSingleClassImplementation()
+            Dim workspace =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document>
+class [|C|] : I { }
+interface $$I { }
+        </Document>
+    </Project>
+</Workspace>
+
+            Test(workspace)
+        End Sub
+
+        <Fact, Trait(Traits.Feature, Traits.Features.GoToImplementation)>
+        Public Sub TestWithTwoClassImplementations()
+            Dim workspace =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document>
+class [|C|] : I { }
+class [|D|] : I { }
+interface $$I { }
+        </Document>
+    </Project>
+</Workspace>
+
+            Test(workspace)
+        End Sub
+
+        <Fact, Trait(Traits.Feature, Traits.Features.GoToImplementation)>
+        Public Sub TestWithOneMethodImplementation()
+            Dim workspace =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document>
+class C : I { public void [|M|]() { } }
+interface I { void $$M(); }
+        </Document>
+    </Project>
+</Workspace>
+
+            Test(workspace)
+        End Sub
+
+        <Fact, Trait(Traits.Feature, Traits.Features.GoToImplementation)>
+        Public Sub TestWithTwoMethodImplementations()
+            Dim workspace =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document>
+class C : I { public void [|M|]() { } }
+class D : I { public void [|M|]() { } }
+interface I { void $$M(); }
+        </Document>
+    </Project>
+</Workspace>
+
+            Test(workspace)
+        End Sub
+
+        <Fact, Trait(Traits.Feature, Traits.Features.GoToImplementation)>
+        Public Sub TestWithNonInheritedImplementation()
+            Dim workspace =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document>
+class C : I { public void [|M|]() { } }
+class D : C, I { }
+interface I { void $$M(); }
+        </Document>
+    </Project>
+</Workspace>
+
+            Test(workspace)
+        End Sub
+
+        <Fact, Trait(Traits.Feature, Traits.Features.GoToImplementation)>
+        Public Sub TestWithInterfaceMemberFromMetdataAtUseSite()
+            Dim workspace =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document>
+using System;
+
+class C : IDisposable
+{
+    public void [|Dispose|]()
+    {
+        IDisposable d;
+        d.$$Dispose();
+    }
+}
+        </Document>
+    </Project>
+</Workspace>
+
+            Test(workspace)
+        End Sub
+    End Class
+End Namespace
```

Comment:
@balajikris: I didn't want the tests to simply retest all the things the downstream APis already take care of. For example, since this is just calling FindSymbolAtPosition we shouldn't need caret tests, or should have the tests testing *that* instead. Which do we think are the tests unique to this?